### ISSUE TITLE
Credentials in constructor

### ISSUE THREAD

**Issue by austinbyers** [carbonblack/cbapi-python#56]
The Python-API [README](https://github.com/carbonblack/cbapi-python/blob/master/README.md) does not include any information about passing options/credentials into the constructor. I thus spent an hour or more trying to get disk credentials to work in a restricted environment.

However, after looking at the code I realized it is in fact possible to pass the `url` and `token` as parameters in the `CbEnterpriseResponseAPI` constructor. **Please update the README to reflect this**

**Comment by jgarman** [comment]
First, I'm sorry that you ended up banging your head against this issue! 

The "original" cbapi had only one way to pass credentials: through the constructor. This led to people uploading scripts to GitHub with their Cb Response global administrator API creds in plain text... bad! To encourage sharing and eliminate the possibility of accidentally exposing your credentials to the Internet, I created the concept of the credential file in cbapi. Note that every other major API client library (for example AWS) uses a similar system to manage their sensitive credentials.

So in a nutshell the omission is intentional, as I wanted to **highly discourage** embedding API tokens inside of scripts.

I will add a note to the CbResponseAPI and CbProtectionAPI constructors in the [readthedocs API reference](https://cbapi.readthedocs.io) so that it's available to those who need it while encouraging the "default" behavior of separating credentials from scripts.

BTW can you share the details of the restricted environment you were working in?

Thanks!

**Comment by austinbyers** [comment]
Thanks for the response! I believe AWS also provides the option to specify the credentials in the CLI.

I'm using CarbonBlack from AWS Lambda, which only lets you write to `/tmp`. Lambda is serverless; the only way to upload a credentials file is to bundle it as part of the deploy package, which is obviously undesirable. Instead, you can pass state through environment variables (encrypted with KMS).

So what I was originally trying to do was take the CarbonBlack credentials from the environment variables and write them to disk at runtime in the expected format. But Lambda will not let you write to any of CarbonBlack's supported locations.

One simple solution would be to allow the CarbonBlack user to specify the location of the credentials file. However, in this case, I don't want the credentials to ever even reach the disk.

Hope this helps!

**Comment by jgarman** [comment]
Awesome, thanks for the info! I'll close this out. The docs have been added to the constructor and I appreciate the description of your environment so I can better design for that use case in the future.
